utils/datasets.py

```diff
@@ -291,6 +291,42 @@ def __len__(self):
         return 0  # 1E12 frames = 32 streams at 30 FPS for 30 years
 
 
+class InfiniteDataLoader(torch.utils.data.dataloader.DataLoader):
+    '''
+    Dataloader that reuses workers.
+
+    Uses same syntax as vanilla DataLoader.
+    '''
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        object.__setattr__(self, 'batch_sampler', _RepeatSampler(self.batch_sampler))
+        self.iterator = super().__iter__()
+
+    def __len__(self):
+        return len(self.batch_sampler.sampler)
+
+    def __iter__(self):
+        for i in range(len(self)):
+            yield next(self.iterator)
```

Comment:
When an epoch is over, we just return the `next` iterator. Therefore, there is no need to reinitialize itself.

Reply:
> Are you keeping an if statement at the end of each batch to check when an epoch would have elapsed and then running all the post-epoch functionality then?

There is no need to check when an epoch ends. Just think of it as literally an Infinite Dataloader that will keep returning batches when you iterate over it. I hope I got what you mean.

Reply:
Ok got it, awesome.